IBX-7418: `ContentName` search criterion added (#2364)
ibexa/documentation-developer: docs/search/criteria_reference/contentname_criterion.md

# ContentName Criterion

The [`ContentName` Search Criterion](https://github.com/ibexa/core/blob/main/src/contracts/Repository/Values/Content/Query/Criterion/ContentName.php)
searches for content by its name.

## Arguments

> - `value` - int(s) representing the Content name

Reviewer: ```suggestion
- `value` - string(s) representing the Content name(s)
```

> - `value` - string representing the content name, the wildcard character `*` can be used for partial search.

Reviewer 2: ```suggestion
- `value` - string representing the content name, the wildcard character `*` can be used for partial search
```

> - `value` - string(s) representing the Content name(s), the wildcard character `*` can be used for partial search.

Reviewer: ```suggestion
- `value` - string representing the Content name, the wildcard character `*` can be used for partial search.
```

## Example

### PHP

``` php
$query->query = new Criterion\ContentName(['laptop', 'tablet']);
```

### REST API

=== "XML"

    ```xml
    <Query>
        <Filter>
            <ContentNameCriterion>laptop</ContentNameCriterion>

>         </Filter>

Reviewer: I may try to have the REST API examples doing the same than the PHP example.

>     <Query>
>         <Filter>
>             <ContentNameCriterion>laptop</ContentNameCriterion>
>         </Filter>
>     </Query>

Reviewer: ```suggestion
    <Query>
        <Filter>
            <OR>
                <ContentNameCriterion>laptop</ContentNameCriterion>
                <ContentNameCriterion>tablet</ContentNameCriterion>
            </OR>
        </Filter>
    </Query>
```

    ```

=== "JSON"

    ```json

>     "Query": {
>         "Filter": {
>             "ContentNameCriterion": "laptop"
>         }
>     }

Reviewer: ```suggestion
    "Query": {
        "Filter": {
            "ContentNameCriterion": ["laptop", "tablet"]
        }
    }
```

# ContentName Criterion

The [`ContentName` Search Criterion](https://github.com/ibexa/core/blob/main/src/contracts/Repository/Values/Content/Query/Criterion/ContentName.php)
searches for content by its name.

## Arguments

> - `value` - string(s) representing the Content name(s), the joker `*` can be used for partial search.

Reviewer 3: ```suggestion
- `value` - string(s) representing the Content name(s), the wildcard character `*` can be used for partial search.
```

It's the first time I see the `joker` name - I thought that wildcard is the common term?

See:
https://learn.microsoft.com/en-us/sql/t-sql/language-elements/percent-character-wildcard-character-s-to-match-transact-sql?view=sql-server-ver16
https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-wildcard-query.html
https://solr.apache.org/guide/8_7/the-standard-query-parser.html#wildcard-searches
https://en.wikipedia.org/wiki/Wildcard_character

## Example

### PHP

``` php

> $query->query = new Criterion\ContentName(['laptop', 'tablet']);

Reviewer 4: Seems the mentioned criterion accepts a single string only: https://github.com/ibexa/core/blob/main/src/contracts/Repository/Values/Content/Query/Criterion/ContentName.php#L16

Reviewer: So, an example illustrting the joker, and that match phone and smart phone.
```suggestion
$query->query = new Criterion\ContentName('*phone');
```

```

### REST API

=== "XML"

    ```xml
    <Query>
        <Filter>

>             <OR>
>                 <ContentNameCriterion>laptop</ContentNameCriterion>
>                 <ContentNameCriterion>tablet</ContentNameCriterion>
>             </OR>

Reviewer: ```suggestion
            <ContentNameCriterion>*phone</ContentNameCriterion>
```

        </Filter>
    </Query>
    ```

=== "JSON"

    ```json
    "Query": {
        "Filter": {

>             "ContentNameCriterion": ["laptop", "tablet"]

Reviewer: ```suggestion
            "ContentNameCriterion": "*phone"
```
